node/overseer/src/lib.rs

```diff
@@ -518,18 +518,36 @@ where
 	}
 	let subsystem_meters = overseer.map_subsystems(ExtractNameAndMeters);
 
-	let memory_stats =
-		MemoryAllocationTracker::new().expect("Jemalloc is the default allocator. qed");
-
-	let metronome = Metronome::new(std::time::Duration::from_millis(950)).for_each(move |_| {
-		match memory_stats.snapshot() {
-			Ok(memory_stats_snapshot) => {
-				tracing::trace!(target: LOG_TARGET, "memory_stats: {:?}", &memory_stats_snapshot);
-				metronome_metrics.memory_stats_snapshot(memory_stats_snapshot);
+	let collect_memory_stats: Box<dyn Fn(&OverseerMetrics) + Send> =
+		match MemoryAllocationTracker::new() {
+			Ok(memory_stats) =>
+				Box::new(move |metrics: &OverseerMetrics| match memory_stats.snapshot() {
+					Ok(memory_stats_snapshot) => {
+						tracing::trace!(
+							target: LOG_TARGET,
+							"memory_stats: {:?}",
+							&memory_stats_snapshot
+						);
+						metrics.memory_stats_snapshot(memory_stats_snapshot);
+					},
+					Err(e) => tracing::debug!(
+						target: LOG_TARGET,
+						"Failed to obtain memory stats: {:?}",
+						e
+					),
+				}),
+			Err(_) => {
+				tracing::debug!(
+					target: LOG_TARGET,
+					"Jemalloc not set as allocator, so memory allocations will not be tracked",
```

Comment:
maybe worth the log message generic if we're to add support for other allocators, but your call :P

Reply:
What you mean by making it generic?

Reply:
Not mentioned Jemalloc

Reply:
Ahh. I mean currently it is only supported for Jemalloc and will probably not change in the near future?

Reply:
But yeah, I remove it